### PR TITLE
Replace AssertDimension

### PR DESCRIPTION
**Pull request by gfcas**
Use ``AssertDimension`` for dimension asserts.

### CODE REVIEW

File: include/deal.II/numerics/vector_tools_interpolate.templates.h [hunk 1]
```diff
@@ -361,10 +360,7 @@ namespace VectorTools
           dof_values.resize(n_dofs);
 
           // Get all function values:
-          Assert(
-            n_components == function(cell)->n_components,
-            ExcDimensionMismatch(dof_handler.get_fe_collection().n_components(),
-                                 function(cell)->n_components));
+          Assert(n_components, function(cell)->n_components);
```
Comment (drwells) [inline include/deal.II/numerics/vector_tools_interpolate.templates.h:363]:
```suggestion
          AssertDimension(n_components, function(cell)->n_components);
```

**Review by drwells — CHANGES_REQUESTED** [review]
You missed one but otherwise 👍

File: include/deal.II/numerics/vector_tools_interpolate.templates.h (hunk 1 above)
Comment (gfcas) [reply include/deal.II/numerics/vector_tools_interpolate.templates.h]: Thanks!

**Review by drwells — APPROVED** [review]

**Pull request merged**